src/Tooltip.jsx

```diff
@@ -47,82 +45,178 @@ class Tooltip extends React.PureComponent {
+    // because the tip ref is needed to calculate the position
+    if (this.props.isOpen) {
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({ isOpen: true });
+    }
   }
 
+  target = null;
   tip = null;
 
+  toggleTip() {
+    this.setState({ showTip: !this.state.showTip });
+  }
+
+  showTip() {
+    this.setState({ showTip: true });
+  }
+
+  hideTip() {
+    this.setState({ hasHover: false });
+    this.setState({ showTip: false });
+  }
+
+  startHover() {
+    this.setState({ hasHover: true });
+
+    setTimeout(this.checkHover, this.props.hoverDelay);
+  }
+
+  endHover() {
+    this.setState({ hasHover: false });
+
+    setTimeout(this.checkHover, this.props.hoverDelay);
+  }
+
+  checkHover() {
+    this.setState({ showTip: this.state.hasHover });
+  }
+
   render() {
     const {
       direction,
-      className,
+      tooltipClassName,
       padding,
+      children,
       content,
+      eventOn,
+      eventOff,
       eventToggle,
       useHover,
       background,
       color,
       useDefaultStyles,
+      isOpen,
       tipContentHover,
       arrow,
       arrowSize,
       distance,
-      target,
+      ...others
     } = this.props;
+    delete others.hoverDelay;
 
-    const currentPositions = positions(direction, this.tip.current, target, {
-      background: useDefaultStyles ? defaultBg : background,
-      arrow,
-      arrowSize,
-      distance,
-    });
+    const showTip = (typeof isOpen === 'undefined') ? this.state.showTip : isOpen;
 
-    const tipStyles = {
-      ...currentPositions.tip,
-      background: useDefaultStyles ? defaultBg : background,
-      color: useDefaultStyles ? defaultColor : color,
-      padding,
-      boxSizing: 'border-box',
-      zIndex: 1000,
-      position: 'absolute',
-      display: 'inline-block',
-    };
+    const props = {};
 
-    const arrowStyles = {
-      ...currentPositions.arrow,
-      position: 'absolute',
-      width: '0px',
-      height: '0px',
-      zIndex: 1001,
-    };
+    // event handling
+    if (eventOff) {
+      props[eventOff] = this.hideTip;
+    }
 
-    const portalProps = {
-      // keep clicks on the tip from closing click controlled tips
-      onClick: stopProp,
-    };
+    if (eventOn) {
+      props[eventOn] = this.showTip;
+    }
+
+    if (eventToggle) {
+      props[eventToggle] = this.toggleTip;
 
-    if (!eventToggle && useHover && tipContentHover) {
-      portalProps.onMouseOver = this.startHover;
-      portalProps.onMouseOut = this.endHover;
-      portalProps.onTouchStart = stopProp;
+      // only use hover if they don't have a toggle event
+    } else if (useHover) {
+      props.onMouseOver = this.startHover;
+      props.onMouseOut = tipContentHover ? this.endHover : this.hideTip;
+      props.onTouchStart = this.toggleTip;
     }
 
+    // should ensure a single child (throws an error instead)
+    React.Children.only(children);
```

Comment:
I'd rather you did this check manually by verifying the children length. That would be much clearer code than this function call that acts like an assertion but is only obvious when reading the comment.